Added build specific test artifact deployment

PhilMFischer opened the pull request:
- SWTBOT Artifacts are now zipped and uploaded to sourceforge
- zip files are individually named, thus they don't get overwritten
- Updated scripts so that maven script can be run locally on Linux computers
- Updated deploy script, so it only deploys in case of test artifacts exist
- Changed strategy on codecov reporting to try to send reports as early as possible. which means also on failed test cases. This may improve finding runtime issues on failing test cases due to race conditions.

Closes #450

ALEhmerDLR committed:
Added blank lines - (Task #411)

added some blank lines in excel und excel.test

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Improved coding style in statemachines.excel- (Task #411)

Added some blank lines
Removed spaces
Renamed 1 letter variables

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Improved coding style in funcelectrial.excel - (Task #411)

Added some blank lines
Removed spaces
Renamed variables

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Improved coding style in .excel - (Task #411)

Added some blank lines
Removed spaces
Renamed variables

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Improved coding style in excel.ui - (Task #411)

Added some blank lines
Removed spaces


---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Changed control flow - (Task #411)

replaced two try-catch with one
Added a variable to show the difference

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Merge remote-tracking branch 'origin/development' into feature/task_411-Improve_coding_style

ALEhmerDLR committed:
# WARNING: head commit changed in the meantime

Changed to e.getMessage() - (Task #411)

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Changed name of variable - (Task #411)

Put your Message here.

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Merge remote-tracking branch 'origin/development' into feature/task_411-Improve_coding_style

ALEhmerDLR committed:
Merged development into this branch - (Task #411)

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Requested Changes - (Task #411)

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Resolved some comments- (Task #411)

Put your Message here.

---
Task #411: Improve coding style in excel plugins

ALEhmerDLR committed:
Renamed and sorted Faults - (Task #411)

Put your Message here.

---
Task #411: Improve coding style in excel plugins

Commit:
Moved env variables from travis to shell script - (Task #450)

- Added the env ariables for overtarget to shell script
- added the env variables for test setup to shell script
- removed variables from tarvis.yml

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Moved env variables from travis to shell script - (Task #450)

- Added the env ariables for overtarget to shell script
- added the env variables for test setup to shell script
- removed variables from tarvis.yml

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Exporting environment variables - (Task #450)

- Setting variables to be exported, thus they are available in the
called maven process

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Exporting environment variables - (Task #450)

- Setting variables to be exported, thus they are available in the
called maven process

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Changed script to upload all swtbot infos in zip - (Task #450)

- Uplaoding zip encoded with the repo job name etc

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Changed script to upload all swtbot infos in zip - (Task #450)

- Uplaoding zip encoded with the repo job name etc

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Force SWTBot to fail- (Task #450)

- Introduce a failure to see if process is working as expected
---
Task #450: Add build specific artefact uplaod to build process

Commit:
Force SWTBot to fail- (Task #450)

- Introduce a failure to see if process is working as expected
---
Task #450: Add build specific artefact uplaod to build process

Commit:
Added download of overtarget to maven script - (Task #450)

- removed setup from tarvis.yml
- added setup to maven shell script

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Added download of overtarget to maven script - (Task #450)

- removed setup from tarvis.yml
- added setup to maven shell script

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixed typo in shell script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixed typo in shell script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Exit without error - (Task #450)

- Added exit 0 instead of 1 or similar
- removed travis retry

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Exit without error - (Task #450)

- Added exit 0 instead of 1 or similar
- removed travis retry

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Exit without error - (Task #450)

- Added exit 0 instead of 1 or similar
- removed travis retry

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Exit without error - (Task #450)

- Added exit 0 instead of 1 or similar
- removed travis retry

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing script - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing script - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Need to set a profile for dependencies - (Task #450)

- Added profile

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Need to set a profile for dependencies - (Task #450)

- Added profile

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Removed cat of ProjectAllTests - (Task #450)

- Project All tests is not generated anymore. Does not need to be
published anymore.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Removed cat of ProjectAllTests - (Task #450)

- Project All tests is not generated anymore. Does not need to be
published anymore.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixed typo - (Task #450)

Put your Message here.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixed typo - (Task #450)

Put your Message here.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

- Finding correct name for deployment zip
- Added debug infos

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

- Finding correct name for deployment zip
- Added debug infos

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

- Finding correct name for deployment zip
- Added debug infos

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

- Finding correct name for deployment zip
- Added debug infos

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Simplifiying travis yaml to speed up testing - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy script - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy shell script - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Fixing deploy shell script - (Task #450)


---
Task #450: Add build specific artefact uplaod to build process

Commit:
Bringing back the original travis yaml - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Bringing back the original travis yaml - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Deploy only if artefacts exist - (Task #450)

- Changed Code to check the directory first before deploying.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Deploy only if artefacts exist - (Task #450)

- Changed Code to check the directory first before deploying.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Be more aggressive ons ending coverage reports - (Task #450)

- Send reports right after maven, This may help spotting issues in case
tests dont run as expected by the use of coverage. Otherwise the reports
are not send at all.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Be more aggressive ons ending coverage reports - (Task #450)

- Send reports right after maven, This may help spotting issues in case
tests dont run as expected by the use of coverage. Otherwise the reports
are not send at all.

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Disable forced failing in swtbot tests - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Disable forced failing in swtbot tests - (Task #450)

---
Task #450: Add build specific artefact uplaod to build process

ALEhmerDLR committed:
Merge remote-tracking branch 'origin/development' into feature/task_411-Improve_coding_style

ALEhmerDLR committed:
Changed the requested Files - (Task #413)

---
Task #413: Remove dead variables in excel export / import classes and
test classes

SaMuellerDLR committed:
Merge pull request #433 from virtualsatellite/feature/task_411-Improve_coding_style

Improve coding style in excel-Plugins

andriikovalov-dlr commented on bash/deploy_sourceforge.sh:
```diff
@@ -33,8 +33,22 @@ printUsage() {
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\//\_/g' <<< ${TEST_ARTEFACTS_ZIP})
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\./\_/g' <<< ${TEST_ARTEFACTS_ZIP}).zip
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
```
Let's remove intermediate debug echoes

PhilMFischer replied:
Yes tried it. Have a look

andriikovalov-dlr commented on bash/deploy_sourceforge.sh:
```diff
@@ -33,8 +33,22 @@ printUsage() {
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\//\_/g' <<< ${TEST_ARTEFACTS_ZIP})
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\./\_/g' <<< ${TEST_ARTEFACTS_ZIP}).zip
```
Can we maybe substitute both / and . in one sed command?

PhilMFischer replied:
Tried it as well. seems to work on a local Linux machine.

andriikovalov-dlr reviewed (changes requested):
I'm wondering if we are deploying all swtbot results then who is going to delete them?

Commit:
Adjustments due to review - (Task #450)

- Removed debug outputs
- Substituted two sed command calls into one

---
Task #450: Add build specific artefact uplaod to build process

Commit:
Adjustments due to review - (Task #450)

- Removed debug outputs
- Substituted two sed command calls into one

---
Task #450: Add build specific artefact uplaod to build process

SaMuellerDLR reviewed (approved):
Approved!

Commit:
Merge branch 'feature/task_450-Add_build_specific_ar' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_450-Add_build_specific_ar
